django_celery_beat/migrations/0001_initial.py

```diff
@@ -71,8 +72,15 @@ class Migration(migrations.Migration):
                     auto_created=True, primary_key=True,
                     serialize=False, verbose_name='ID')),
                 ('name', models.CharField(
-                    help_text='Useful description', max_length=200,
-                    unique=True, verbose_name='name')),
+                    help_text='Useful description',
+                    max_length=getattr(
+                        settings,
+                        'DJANGO_CELERY_BEAT_NAME_MAX_LENGTH',
```

Comment:
Whoa...  you can't base migrations off of a settings option that people can change, that makes things completely unpredictable.  Just choose a more correct value, one that won't ever have the issue again.

Reply:
> Whoa...

Can you elaborate on your surprise?

1. The code is not "unpredictable", the defaults are backwards compatible
2. Projects with even more stars follow this route (https://github.com/python-social-auth/social-app-django/blob/master/social_django/migrations/0001_initial.py)
3. Simply changing this to 191 would solve it for me, but everyone using this with e.g. postgresql would be facing a DB schema which differs from the migrations

Reply:
@liquidpele can't really choose "a more correct value" in this case since some values work for some databases and the same values cause errors in other databases. 

As @tuky points out driving the max_length value from the settings is not a new pattern and its pretty well-established.

If there is another approach you have in mind, please share.

Reply:
> Whoa... you can't base migrations off of a settings option that people can change, that makes things completely unpredictable. Just choose a more correct value, one that won't ever have the issue again.

IMHO this is allowed in some case... but we need to handle this with care. let us not rush on merging this.

Reply:
It's not that some values work and some don't, it's simply a limitation in mysql, one that is already fixed in newer versions where they increased the prefix limit for innodb tables.  

One question I have, and @auvipy may know this... is the task_id ever NOT a uuid?  I thought it was always a uuid, but a length of 255 there makes me think it could be other things too?

Reply:
I do somehow agree with you. Need some analysis to answer your question properly